ElectronicObserver/Data/Battle/BattleRankPrediction.cs

```diff
@@ -0,0 +1,287 @@
+
+		for (int index = 0; index < FriendlyMainFleetBefore.MembersWithoutEscaped.Count; index++)
+		{
+			int? hpBefore = FriendlyMainFleetBefore.MembersWithoutEscaped[index]?.HPCurrent;
+			if (hpBefore is null or < 0) continue;
+
+			int? hpAfter = FriendlyMainFleetAfter.MembersWithoutEscaped[index]?.HPCurrent;
+			if (hpAfter is null) continue;
+
+			FriendlyHpBefore += (int)hpBefore;
+			FriendlyHpAfter += Math.Max((int)hpAfter, 0);
+			FriendlyShipCount++;
+
+			if (hpAfter <= 0)
+			{
+				FriendlyShipSunk++;
+			}
+		}
+	}
+
+	private void CalculeFriendlyEscortFleetHp()
+	{
+		if (FriendlyEscortFleetBefore?.MembersWithoutEscaped is null) return;
+		if (FriendlyEscortFleetAfter?.MembersWithoutEscaped is null) return;
+
+		for (int index = 0; index < FriendlyEscortFleetBefore.MembersWithoutEscaped.Count; index++)
+		{
+			int? hpBefore = FriendlyEscortFleetBefore.MembersWithoutEscaped[index]?.HPCurrent;
+			if (hpBefore is null or < 0) continue;
+
+			int? hpAfter = FriendlyEscortFleetAfter.MembersWithoutEscaped[index]?.HPCurrent;
+			if (hpAfter is null) continue;
+
+			FriendlyHpBefore += (int)hpBefore;
+			FriendlyHpAfter += Math.Max((int)hpAfter, 0);
+			FriendlyShipCount++;
+
+			if (hpAfter <= 0)
+			{
+				FriendlyShipSunk++;
+			}
+		}
+	}
+
+	private void CalculateEnemyMainFleetHp()
+	{
+		for (int index = 0; index < EnemyMainFleetBefore.MembersInstance.Count; index++)
+		{
+			IShipData? ship = EnemyMainFleetBefore.MembersInstance[index];
+
+			if (ship is null) continue;
+			if (!ship.CanBeTargeted) continue;
+
+			int hpBefore = ship.HPCurrent;
+			if (hpBefore < 0) continue;
+
+			int? hpAfter = EnemyMainFleetAfter.MembersInstance[index]?.HPCurrent;
+			if (hpAfter is null) continue;
+
+			EnemyHpBefore += hpBefore;
+			EnemyHpAfter += Math.Max((int)hpAfter, 0);
+			EnemyShipCount++;
+
+			if (hpAfter <= 0)
+			{
+				EnemyShipSunk++;
+			}
+		}
+	}
+
+	private void CalculateEnemyEscortFleetHp()
+	{
+		if (EnemyEscortFleetBefore is null) return;
+		if (EnemyEscortFleetAfter is null) return;
+
+		for (int index = 0; index < EnemyEscortFleetBefore.MembersInstance.Count; index++)
+		{
+			IShipData? ship = EnemyEscortFleetBefore.MembersInstance[index];
+
+			if (ship is null) continue;
+			if (!ship.CanBeTargeted) continue;
+
+			int hpBefore = ship.HPCurrent;
+			if (hpBefore < 0) continue;
+
+			int? hpAfter = EnemyEscortFleetAfter.MembersInstance[index]?.HPCurrent;
+			if (hpAfter is null) continue;
+
+			EnemyHpBefore += hpBefore;
+			EnemyHpAfter += Math.Max((int)hpAfter, 0);
+			EnemyShipCount++;
+
+			if (hpAfter <= 0)
+			{
+				EnemyShipSunk++;
+			}
+		}
+	}
+
+	private void ResetValues()
+	{
+		FriendlyShipCount = 0;
+		FriendlyShipSunk = 0;
+		FriendlyHpBefore = 0;
+		FriendlyHpAfter = 0;
+
+		EnemyShipCount = 0;
+		EnemyShipSunk = 0;
+		EnemyHpBefore = 0;
+		EnemyHpAfter = 0;
+	}
+
+	/// <summary>
+	/// Create a copy of the fleet with their HP after the battle
+	/// </summary>
+	/// <param name="fleet"></param>
```

Comment:
You can just remove the param and returns parts if there's no extra description needed.